config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/beans/Policy.java

```diff
@@ -227,21 +227,21 @@ public void postLoad(String entityKey, Bundle bundle, @Nullable File rootFolder,
         setPath(PathUtils.unixPath(getPath()));
     }
 
-    public boolean hasAnnotated() {
+    public boolean hasAnnotation() {
         return annotations != null && !annotations.isEmpty();
     }
 
     @Override
     public boolean hasBundleAnnotation(){
-        if(hasAnnotated()){
-            return annotations.stream().anyMatch(annotation -> ANNOTATION_TYPE_BUNDLE.equals(annotation.getType()));
+        if (hasAnnotation()){
+            return annotations.contains(new Annotation(ANNOTATION_TYPE_BUNDLE));
```

Comment:
Instead of duplicating this code, we can have an `interface` with default method implementation and have all those class which can be annotated `implement` that interface.